vm/statement_test.go

```diff
@@ -156,6 +156,20 @@ func TestDefStatement(t *testing.T) {
 
 		foo(10, 20, 30)
 		`, 60},
+		{`
+		def foo(a:20)
+		  a
+		end
+
+		foo(a:10)
+		`, 10},
+		{`
+		def foo(a:20)
+		  a
+		end
+
+		foo
+		`, 20},
```

Comment:
@st0012 I'm kind of stuck here... The last test does not pass, and I believe I set default value at `line 178 - 192 in satement_generation.go`, but it raises out of index error probably at `line 281 in thread.go`. Could you help me out here?

Also, if you have any adviser for overall implementation, I'm glad to here that, too! (Although 40% of implementation has done so far I feel..)

Reply:
Ok, I just noticed I should not call `t.stack.Data[argPr+argIndex]` when default value is not give! So, I kind of solved this problem, but still I'd appreciate it if would give me an advise! haha

Reply:
BTW, I'm aware that my current implementation depends on the order of arguments, so `test(a: 1, b:2)` and `test(b: 2, a: 1)` give a different result. I'm going to work on the problem next.